classes/class-p4-master-site.php

```diff
@@ -1103,4 +1106,34 @@ public function add_help_sidebar() {
 			$screen->set_help_sidebar( $sidebar );
 		}
 	}
+
+	/**
+	 * Generate structured data for a search engine.
+	 */
+	public function generate_p4_structured_data() {
+		$p4_site_url   = get_site_url();
+		$p4_site_title = get_bloginfo( 'name' );
+
+		// Filter social menu links.
+		$footer_social_menu = wp_get_nav_menu_items( 'Footer Social' );
+		$social_menu_links  = [];
+
+		foreach ( $footer_social_menu as $menu_details ) {
```

Comment:
Please check also the failed unit tests, here should be a check that there are indeed menu items for example